sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/v2/state/StatePartitionReader.scala

```diff
@@ -238,7 +238,23 @@ class StateStoreChangeDataPartitionReader(
   }
 
   override lazy val iter: Iterator[InternalRow] = {
-    changeDataReader.iterator.map(unifyStateChangeDataRow)
+    if (stateVariableInfoOpt.isDefined && stateVariableInfoOpt.get.stateVariableType ==
```

Comment:
nits: we can reuse the function: `checkVariableType` [here](https://github.com/apache/spark/blob/master/sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/v2/state/utils/SchemaUtil.scala#L342)?

Reply:
Done